crates/nargo_cli/tests/test_data_ssa_refactor/slices/src/main.nr

```diff
@@ -0,0 +1,21 @@
+use dep::std::slice;
+
+fn main(x : Field, y : pub Field) {
+
+    let mut slice: [Field] = [0; 2];
+
+    assert(slice[0] == 0);
+    assert(slice[0] != 1);
+
+    let slice_plus_10 = slice.push_back(10);
+    assert(slice_plus_10[2] == 10);
+    assert(slice_plus_10[2] != 8);
+    assert(slice_plus_10.len() == 3);
+
+    let mut new_slice: [Field] = [];
```

Comment:
You could possibly allow both, and error if something non-Rusty were to happen (so probably it'd be immutable if used with &)

Reply:
The issue is more of a type issue. If we see `[a, b, c]` and both arrays and slices use the syntax, we don't know which type to infer. In some cases it may be fine to infer the more narrow array type is used, but if mutability is involved like in the loop in this example, we wouldn't be able to call `new_slice = new_slice.push_back(...)` in the loop since we'd be assigning a slice type to an array.

Reply:
One solution is to say we don't need arrays at all if we have slices and get rid of the array type entirely. Sized arrays would only be as parameters to main like `pub` is. This way all `[]` literals in the code would have a slice type.